### PR TITLE
Mccalluc/version awareness

### PR DESCRIPTION
**Pull request by mccalluc**
- Towards #2111

@shirey or @yuanzhou , can you point me at a couple examples of versioned entities, in any environment?

### CODE REVIEW

**Comment by yuanzhou** [comment]
TEST: https://search-api.test.hubmapconsortium.org/status
PROD: https://search.api.hubmapconsortium.org/status

In the resulting json, the `build` number indicates the git branch name and commit hash of the codebase. The `version` number is the release/tag.

**Comment by mccalluc** [comment]
> In the resulting json, the `build` number indicates the git branch name and commit hash of the codebase. The `version` number is the release/tag.

@yuanzhou -- Sorry for the misunderstanding. I'm asking about the versioning of entities, in particular datasets, not the version numbers of services. We're already using that API for our [services page](https://portal.hubmapconsortium.org/services).

File: context/app/static/js/hooks/useSearchData.js [hunk 1]
```diff
@@ -7,12 +7,31 @@ function useSearchData(query) {
...
+  }
+  if (!('exists' in queryCurrent.bool.must_not)) {
+    queryCurrent.bool.must_not.exists = {};
+  }
+  if (!('field' in queryCurrent.bool.must_not.exists)) {
+    queryCurrent.bool.must_not.exists.field = 'next_revision_uuid';
+  } else {
+    throw new Error(`Bug: bool.must_not.exists.field already set: ${query}`);
+  }
+  /* eslint-enable prettier/prettier */
```
Comment (mccalluc) [inline context/app/static/js/hooks/useSearchData.js:27]:
```suggestion
  /* eslint-disable prettier/prettier */
  if (!('bool'     in queryCurrent))                      queryCurrent.bool = {};
  if (!('must_not' in queryCurrent.bool))                 queryCurrent.bool.must_not = {};
  if (!('exists'   in queryCurrent.bool.must_not))        queryCurrent.bool.must_not.exists = {};
  if (!('field'    in queryCurrent.bool.must_not.exists)) queryCurrent.bool.must_not.exists.field = 'next_revision_uuid';
  else {
    throw new Error(`Bug: bool.must_not.exists.field already set: ${query}`);
  }
  /* eslint-enable prettier/prettier */
```

I had wanted to commit something more like this, but it got auto-formatted.
There are utils out there to set deep properties... are we already using one in our codebase?

**Comment by shirey** [comment]
More examples on TEST:
 V1: 2dc0514607ae08af7004d9cb8c9640c9 (Published/public)
 V2: 0337c0b973273b677c5ec9a31eb33983 (Published/public)
 V3: a20e6d2cfeba9bdc7cd56573b7492320 (Published/public)
 V4: 98e760c2af3d221a4ce6e2e012110f37 (Published/public)
 V5: 72605b56bc5c452a3b6948e4b6b983ed (Published/public)
 
 V1: a9e97dfb17c7fe08a9c4590318955095 (Published/public)
 V2: 437536cb0fae9e6a9be0ea3670fa69db (status:QA)

**Comment by mccalluc** [comment]
John has handled this separately.

**Pull request closed without merging**